scionlab/tests/test_user_as_models.py

```diff
@@ -123,69 +139,80 @@ def create_and_check_useras(testcase,
-                 bind_port,
+                 vpn_choice,
                  installation_type,
                  label):
     """
-    Check the state of `user_as` and `attachment_point`.
-    Verify that a link to the attachment point exists and that it is configured according to the
-    given parameters.
+    Check the state of `user_as` and `aps_confs`.
+
+    Verify that the links to the attachment points exists and that they are configured according to
+    the given parameters.
     """
     testcase.assertEqual(user_as.owner, owner)
     testcase.assertEqual(user_as.label, label)
     testcase.assertEqual(user_as.installation_type, installation_type)
-    testcase.assertEqual(user_as.attachment_point, attachment_point)
-    testcase.assertEqual(user_as.is_use_vpn(), use_vpn)
-    testcase.assertEqual(user_as.public_ip, public_ip)
-    testcase.assertEqual(user_as.get_public_port(), public_port)
-    testcase.assertEqual(user_as.bind_ip, bind_ip)
-    testcase.assertEqual(user_as.bind_port, bind_port)
-
     utils.check_as(testcase, user_as)
-    utils.check_as(testcase, attachment_point.AS)
-
-    link = get_provider_link(attachment_point.AS, user_as)
-    if use_vpn:
-        utils.check_link(testcase, link, utils.LinkDescription(
-            type=Link.PROVIDER,
-            from_as_id=attachment_point.AS.as_id,
-            from_public_ip=attachment_point.vpn.server_vpn_ip,
-            from_bind_ip=None,
-            from_internal_ip=DEFAULT_HOST_INTERNAL_IP,
-            to_public_ip=user_as.hosts.get().vpn_clients.get(active=True).ip,
-            to_public_port=public_port,
-            to_bind_ip=None,
-            to_bind_port=None,
-            to_internal_ip=DEFAULT_HOST_INTERNAL_IP,
-        ))
-    else:
-        overridden_bind_ip = bind_ip
-        if installation_type == UserAS.VM:
-            overridden_bind_ip = '10.0.2.15'
-        overridden_bind_port = bind_port
-        if overridden_bind_ip:
-            overridden_bind_port = bind_port or public_port
-
-        utils.check_link(testcase, link, utils.LinkDescription(
-            type=Link.PROVIDER,
-            from_as_id=attachment_point.AS.as_id,
-            from_public_ip=_get_public_ip_testtopo(attachment_point.AS.as_id),
-            from_bind_ip=None,
-            from_internal_ip=DEFAULT_HOST_INTERNAL_IP,
-            to_public_ip=public_ip,
-            to_public_port=public_port,
-            to_bind_ip=overridden_bind_ip,
-            to_bind_port=overridden_bind_port,
-            to_internal_ip=DEFAULT_HOST_INTERNAL_IP,
-        ))
-
-    _check_attachment_point(testcase, attachment_point)
+
+    # Check that the AttachmentPoints in `aps_confs` are now AttachmentPoints of the user_as
+    aps_ases = [c.attachment_point.AS for c in aps_confs]
+    user_as_aps_ases = [l.interfaceA.AS
+                        for l in Link.objects.filter(interfaceB__AS=user_as).all()]
+    testcase.assertEqual(set(user_as_aps_ases), set(aps_ases))
```

Comment:
This should not (only) compare sets, entries can have multiplicities. Compare sorted lists instead.